python_ta/transforms/type_inference_visitor.py

```diff
@@ -252,8 +258,9 @@ def register_type_constraints_setter():
     return type_visitor
 
 
-def _set_locals_environment(node):
-    node.type_environment = Environment(locals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.locals})
+def _set_environment(node):
+    node.type_environment = Environment(locals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.locals},
+                                        globals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.globals})
```

Comment:
You can set up a quick script to run python_ta.check_all() on some example files to ensure no errors (particularly this AttributeError).